dali/dfu/dfuutil.cpp

```diff
@@ -1062,6 +1062,31 @@ class CDFUhelper: implements IDFUhelper, public CInterface
         else if (0 == strcmp(nodeName, queryDfsXmlBranchName(DXB_File)))
         {
             // Logical file map
+            //Patch the cluster before the file is created
+            if (!isEmptyString(cluster))
+            {
+                //In containerized mode the directory is configured somewhere else...
+                //replace @directory with a relative directory calculated from the scope
+                //MORE: This could should be commoned up with similar code elsewhere
+                Owned<IStoragePlane> plane = getStoragePlane(cluster, true);
+                StringBuffer location(plane->queryPrefix());
+                const char * temp = lfn;
+                for (;;)
+                {
+                    const char * sep = strstr(temp, "::");
+                    if (!sep)
+                        break;
+                    addPathSepChar(location);
+                    location.appendLower(sep - temp, temp);
+                    temp = sep + 2;
+                }
+                t->setProp("@directory", location);
+
+                //Remove any clusters in the incoming definition, and replace it with a single entry
+                t->removeProp("Cluster");
```

Comment:
to remove all "Cluster"'s you'd need : 
while (t->removeProp("Cluster"));

Reply:
@jakesmith actually I think the original code is correct - the ptree code removes all items that match the property.

Reply:
```
    Owned<IPropertyTree> pt = createPTree("me");
    pt->addPropTree("b", createPTree());
    pt->addPropTree("b", createPTree());
    pt->addPropTree("c", createPTree());
    pt->addPropTree("d", createPTree());
    printXML(pt);
    pt->removeProp("b");
    printXML(pt);


```

Reply:
hmmm - that's a bug.
If you remove a property with a qualifier or with a wildcard, it removes 1st match.
I see from the impl. though if it's a simple id, it removes the entry from the children, which may be a list container.

It should remove the 1st. To remove all there should be a loop, as there are elsewhere.

I still recommend changing it to while(.. removeProp ..) even if it is spuriously removing all on the 1st attempt.

I'll open a JIRA to look at the semantic difference, but am concerned re. changing it.. what will it break.
Further discussion in that JIRA.

Reply:
https://track.hpccsystems.com/browse/HPCC-25281